File: js/examples/styles.js
const { Tracer } = require('../tracer.js');

async function simulateWork(tracer) {
    tracer.span('Load configuration');
    await new Promise(resolve => setTimeout(resolve, 25));

    tracer.span('Connect to database');
    await new Promise(resolve => setTimeout(resolve, 75));

    tracer.span('Execute complex query');
    await new Promise(resolve => setTimeout(resolve, 120));

    tracer.span('Process results');
    await new Promise(resolve => setTimeout(resolve, 45));

    tracer.span('Cache data');
    await new Promise(resolve => setTimeout(resolve, 15));

    tracer.span('Generate response');
    await new Promise(resolve => setTimeout(resolve, 8));
}

async function main() {
    const styles = [
        { name: 'Default', style: 'default' },
        { name: 'Colorful', style: 'colorful' },
        { name: 'Minimal', style: 'minimal' },
        { name: 'Detailed', style: 'detailed' },
        { name: 'Table', style: 'table' },
        { name: 'JSON', style: 'json' }
    ];

    for (const s of styles) {
        console.log(`\n\x1b[1;36m=================== ${s.name.toUpperCase()} STYLE ===================\x1b[0m`);

        const tracer = new Tracer(`API Processing - ${s.name} Style`, {
            style: s.style
        });

        await simulateWork(tracer);
        tracer.end();

        await new Promise(resolve => setTimeout(resolve, 500)); // Brief pause between styles
    }
}

main().catch(console.error); 

File: go/examples/styles.go
//go:build ignore
// +build ignore

package main

import (
	"fmt"
	"time"

	tracing "github.com/LeDuyViet/quicktrace/go"
)

func simulateWork(tracer *tracing.Tracer) {
	time.Sleep(25 * time.Millisecond)
	tracer.Span("Load configuration")

	time.Sleep(300 * time.Millisecond)
	tracer.Span("Connect to database")

	time.Sleep(1000 * time.Millisecond)
	tracer.Span("Execute complex query")

	time.Sleep(100 * time.Millisecond)
	tracer.Span("Process results")

	time.Sleep(150 * time.Millisecond)
	tracer.Span("Cache data")

	time.Sleep(1000 * time.Millisecond)
	tracer.Span("Execute complex query 2")

	time.Sleep(50 * time.Millisecond)
	tracer.Span("Generate response")
}

func main() {
	styles := []struct {
		name  string
		style tracing.OutputStyle
	}{
		{"Default", tracing.StyleDefault},
		{"Colorful", tracing.StyleColorful},
		{"Minimal", tracing.StyleMinimal},
		{"Detailed", tracing.StyleDetailed},
		{"Table", tracing.StyleTable},
		{"JSON", tracing.StyleJSON},
	}

	for _, s := range styles {
		fmt.Printf("\n%s=================== %s STYLE ===================%s\n",
			"\033[1;36m", s.name, "\033[0m")

		tracer := tracing.NewSimpleTracer("API Processing - "+s.name+" Style",
			tracing.WithOutputStyle(s.style))

		simulateWork(tracer)
		tracer.End()

		time.Sleep(500 * time.Millisecond) // Brief pause between styles
	}
}
